src/v/storage/segment_appender.h

```diff
@@ -49,7 +51,8 @@ class segment_appender {
 
         ss::io_priority_class priority;
         size_t number_of_chunks;
-        size_t falloc_step{fallocation_step};
+        size_t falloc_step{
+          config::shard_local_cfg().segment_fallocation_step()};
```

Comment:
we need to stop using the thread-local for this stuff but pass it as an option  on the caller, all the way to main. 

testing core-local configs are super hard. generally not a good practice to use it for tests in particular.